src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -442,43 +515,98 @@ private static BidRequest givenBidRequest(
+                        givenSeatBid(
+                                seatBid -> seatBid.seat("seat2"),
+                                bid -> bid.id("bid2").price(BigDecimal.valueOf(0.5)))))
                 .cur("USD")).build();
     }
 
-    private static SeatBid givenSeatBid(UnaryOperator<SeatBid.SeatBidBuilder> seatBidCostumizer) {
+    private static SeatBid givenSeatBid(UnaryOperator<SeatBid.SeatBidBuilder> seatBidCostumizer,
+                                        UnaryOperator<Bid.BidBuilder>... bidCustomizers) {
         return seatBidCostumizer.apply(SeatBid.builder()
-                .bid(Collections.singletonList(givenBid(bid -> bid)))
-                .seat("seat1")).build();
+                        .bid(givenBids(bidCustomizers))).build();
+    }
+
+    private static List<Bid> givenBids(UnaryOperator<Bid.BidBuilder>... bidCustomizers) {
+        return Arrays.stream(bidCustomizers)
+                .map(GreenbidsAnalyticsReporterTest::givenBid)
+                .toList();
     }
 
     private static Bid givenBid(UnaryOperator<Bid.BidBuilder> bidCustomizer) {
         return bidCustomizer.apply(Bid.builder()
-                .id("bid1")
                 .impid("adunitcodevalue")
-                .price(BigDecimal.valueOf(1.5))
                 .adm("<div>Ad Markup</div>")).build();
     }
 
     private static BidRejectionTracker givenBidRejectionTracker() {
         final BidRejectionTracker bidRejectionTracker = new BidRejectionTracker(
-                "seat2",
+                "seat3",
                 Set.of("adunitcodevalue"),
                 1.0);
         bidRejectionTracker.reject("imp1", BidRejectionReason.NO_BID);
         return bidRejectionTracker;
     }
 
+    private static ObjectNode givenPrebidBidderParamsNode() {
+        final ObjectNode bidderNode = mapper.createObjectNode();
+
+        final ObjectNode seat1Params = mapper.createObjectNode()
+                .put("accountId", 1001)
+                .put("siteId", 267318)
+                .put("zoneId", 1861698);
+        bidderNode.set("seat1", seat1Params);
+
+        final ObjectNode seat2Params = mapper.createObjectNode()
+                .put("publisherId", 111)
+                .put("adSlotId", 123456);
+        bidderNode.set("seat2", seat2Params);
+
+        final ObjectNode seat3Params = mapper.createObjectNode()
+                .put("placementId", 222);
+        bidderNode.set("seat3", seat3Params);
+
+        final ObjectNode prebidNode = mapper.createObjectNode();
+        prebidNode.set("bidder", bidderNode);
+
+        return prebidNode;
+    }
+
+    private static UnaryOperator<ObjectNode> createBidderParams(
+            String bidder, UnaryOperator<ObjectNode> paramsCustomizer) {
+        return bidderNode -> {
+            final ObjectNode paramsNode = mapper.createObjectNode();
+            paramsCustomizer.apply(paramsNode);
+            bidderNode.set(bidder, paramsNode);
+            return bidderNode;
+        };
+    }
```

Comment:
still returning the operator

Reply:
ah thanks, I've removed unused createBidderParams